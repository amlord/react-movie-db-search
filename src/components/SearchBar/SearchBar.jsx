import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledSearchBar = styled.div`
  background-color: #01d277;
  color: #333;
  border-radius: 4px;
  padding: 5px;
  margin-bottom: 20px;
  display: grid;
  grid-template-columns: 120px auto;
`;

const StyledLabel = styled.label`
  font-size: 1em;
  line-height: 25px;
  padding-left: 5px;
`;

const StyledInput = styled.input`
  border-radius: 2px;
  font-size: 1em;
  line-height: 25px;
  border: none;
  padding: 0 5px;
`;

const SearchBar = ({ onSearchUpdate }) => (
  <StyledSearchBar>
    <StyledLabel htmlFor="search">Search Movies:</StyledLabel>
    <StyledInput
      type="search"
      id="search"
      onChange={event => onSearchUpdate(event.target.value)}
    />
  </StyledSearchBar>
);

SearchBar.propTypes = {
  onSearchUpdate: PropTypes.func.isRequired,
};

export default SearchBar;
