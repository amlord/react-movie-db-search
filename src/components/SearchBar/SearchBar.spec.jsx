import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from './SearchBar';

describe('SearchBar', () => {
  it('displays correctly', () => {
    const wrapper = shallow(<SearchBar onSearchUpdate={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });
});
