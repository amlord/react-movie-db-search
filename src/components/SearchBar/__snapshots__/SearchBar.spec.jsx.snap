// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar displays correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    onSearchUpdate={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <SearchBar__StyledLabel
          htmlFor="search"
        >
          Search Movies:
        </SearchBar__StyledLabel>,
        <SearchBar__StyledInput
          id="search"
          onChange={[Function]}
          type="search"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Search Movies:",
          "htmlFor": "search",
        },
        "ref": null,
        "rendered": "Search Movies:",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "search",
          "onChange": [Function],
          "type": "search",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <SearchBar__StyledLabel
            htmlFor="search"
          >
            Search Movies:
          </SearchBar__StyledLabel>,
          <SearchBar__StyledInput
            id="search"
            onChange={[Function]}
            type="search"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Search Movies:",
            "htmlFor": "search",
          },
          "ref": null,
          "rendered": "Search Movies:",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "search",
            "onChange": [Function],
            "type": "search",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
